sample-plugin-candlestick-chart-new/src/CandlestickChart.js

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { format } from "d3-format";
+import { ChartCanvas, Chart } from "react-stockcharts";
+import { CandlestickSeries } from "react-stockcharts/lib/series";
+import { XAxis, YAxis } from "react-stockcharts/lib/axes";
+import {
+	CrossHairCursor,
+	EdgeIndicator,
+	MouseCoordinateY,
+} from "react-stockcharts/lib/coordinates";
+
+import { discontinuousTimeScaleProvider } from "react-stockcharts/lib/scale";
+import { OHLCTooltip } from "react-stockcharts/lib/tooltip";
+import { ema, stochasticOscillator, bollingerBand } from "react-stockcharts/lib/indicator";
+import { fitWidth } from "react-stockcharts/lib/helper";
+import { last } from "react-stockcharts/lib/utils";
+
+class CandlestickChart extends React.Component {
```

Comment:
We should be using functional components for these examples since we want our customers to see we are keeping up to date with modern react practices. Plus we want people to practice good practices if they copy our examples. Let me know if you need help converting the component.